src/electionguard/decryption_mediator.py

```diff
@@ -84,34 +84,38 @@ def announce(
         """
 
         # Only allow a guardian to announce once
-        if guardian.object_id in self._available_guardians:
-            log_info(f"guardian {guardian.object_id} already announced")
+        if guardian.id in self._available_guardians:
+            log_info(f"guardian {guardian.id} already announced")
             return (
-                self._tally_shares[guardian.object_id],
-                self._ballot_shares[guardian.object_id],
+                self._tally_shares[guardian.id],
+                self._ballot_shares[guardian.id],
             )
 
         # Compute the tally and ballot decryption shares
         tally_share = compute_decryption_share(
-            guardian, self._ciphertext_tally, self._encryption
+            guardian._election_keys,  # pylint: disable=protected-access
```

Comment:
Private keys are referenced here in the tests but this is not an optimal case. Here we are using the stateful methods to quickly construct guardians but this method is a meant to be used in a functional capacity.